src/devices/chatBot_nws_yarp/ChatBotRPC_CallbackHelper.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * SPDX-FileCopyrightText: 2023-2023 Istituto Italiano di Tecnologia (IIT)
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "ChatBot_nws_yarp.h"
+
+#include <yarp/os/LogComponent.h>
+#include <yarp/os/LogStream.h>
+#include <yarp/os/Os.h>
+
+namespace {
+YARP_LOG_COMPONENT(CHATBOTRPC_HELPER, "yarp.devices.chatBot_nws_yarp.ChatBotRPC_CallbackHelper")
+}
+
+ChatBotRPC_CallbackHelper::ChatBotRPC_CallbackHelper(yarp::dev::IChatBot* iChat, yarp::os::Port* p)
+{
+    if (iChat ==nullptr || p==nullptr)
+    {
+        yCError(CHATBOTRPC_HELPER) << "Could not get IChatBot interface/output port";
+        std::exit(1);
```

Comment:
maybe used also elsewhere, but could be improved

Reply:
I modified the structure of the class a little bit, in order to avoid the exit:
[link](https://github.com/elandini84/yarp/blob/32243697b709741f1d72496acc8d8db68dd1d48c/src/devices/chatBot_nws_yarp/ChatBotRPC_CallbackHelper.cpp#L16-L27)